packages/batch-delegate/tests/cacheByArguments.test.ts

```diff
@@ -110,4 +119,66 @@ describe('non-key arguments are taken into account when memoizing result', () =>
     expect(chirps[0].withoutObfuscatedEmail.email).toBe(`1@test.com`);
     expect(chirps[1].withoutObfuscatedEmail.email).toBe(`2@test.com`);
   });
+  test('memoizes key arguments as part of batch delegation', async () => {
```

Comment:
can we expand the tests to cover more cases?

like more nested levels?
or more then two?

Reply:
Good point! Added one tests deeper levels...

Reply:
thanks!
what about if the arguments are in deeper nested levels and not on the root level?